snapcraft/plugins/catkin.py

```diff
@@ -348,36 +348,65 @@ def _prepare_build(self):
         # dependencies contain .cmake files pointing to system paths (e.g.
         # /usr/lib, /usr/include, etc.). They need to be rewritten to point to
         # the install directory.
-        def rewrite_paths(match):
+        def _new_path(path):
+            if not path.startswith(self.installdir):
+                return self.installdir + path
```

Comment:
This is actually because the `path` here is absolute. If I were to use `os.path.join` I'd have to do `path[1:]`. Is that preferable?